crates/toasty-codegen/src/schema/field.rs

```diff
@@ -157,6 +158,18 @@ impl Field {
             ));
         }
 
+        // Validate column type compatibility for primitive fields
+        if ty.is_none() && attrs.column.is_some() {
```

Comment:
```suggestion
        if ty.is_none() {
```

`attires.column.is_some()` is checked in `validate_column_type_compatibility`.